packages/compiler/core/checker.ts

```diff
@@ -5121,6 +5121,20 @@ export function getEffectiveModelType(
 
   if (model.properties.size === 0) {
     // empty model
+    switch (model.node?.kind) {
+      // workaround where we have an intersection with no resulting properties. Take the name of the canonical thing,
+      // which we assume to be the first type reference in the intersection.
+      case SyntaxKind.IntersectionExpression:
+        for (const opt of model.node.options) {
+          if (opt.kind === SyntaxKind.TypeReference && opt.target.kind === SyntaxKind.Identifier) {
+            model.name = opt.target.sv;
+            break;
+          }
+        }
+        break;
+      default:
+        return model;
+    }
```

Comment:
@johanste this is a icky workaround and it fails if the canonical model has no properties but there were, for example, traits.

Reply:
Tagging myself to review: @nguerrera

Reply:
replying for your 4 different  use cases:

**1. Intersection has no properties**
This one I think is kinda wrong unless we agree that if it find multiple choices it takes the first one
**2. Intersection has properties from Canonical but not Extra**
Feels right 
**3. Intersection has properties from Extra but not Canonical**
This one I think is correct because extra is with no metadata properties
**4. Intersection has properties from Extra and Canonical**
this feels correct to me, if extra had only metadata properties then I would expect it to return `Cannonical` but here there is both models that have value
 

I think with the introduction of empty models we have to answer those question: 
- What happen if there is multiple empty models that are good options. Do we take the first one?
- If there is an empty model + another model that has properties that we keep. Do we keep the other model?

Reply:
+1 

Additional consideration: I don't think this should be tied specifically to intersection. I think `{ ...Canonical, ...Extra }` should behave the same as `Canonical & Extra`.

Reply:
Typescript has this `constituentMap` in `UnionOrIntersectionType` wondering if we can have the same thing instead of depending on the intersection.

Reply:
It seems reasonable to have something like `sourceModel` for `is` that is inclusive of spread and intersection that could tell you given a model which models were `is`-ed, spread, or intersected into this model, which would provide a way to deal with empty model spread/is/intersect without going to source node. We still have to decide the right behaviors for empty model.

Reply:
Right now we treat all of those things as equal (`A & B` or `...A, ...B`). What if we somehow expressed which thing, if any, were the most important? In this case, a way to say "Canonical is the most important thing" so that in the case 4, it would generate some kind of "CanonicalResponse" or something that combines the two models instead of just "(anonymous model)" with no schema. Maybe this already happens on the code generation side.

Reply:
problem with this here is you cannot apply a decorator to an expression so wouldn't work without a change in the compiler(and most likely a big design discussion)

however in the same way we could have the decorator on teh operation

```ts
@effectiveModel(TParams)
op RpcOperation<
  TParams extends TypeSpec.Reflection.Model,
```

other problem with this is now you need to watch for both system

Reply:
What do you mean watch for both systems?

Reply:
depending on where this decorator is  it might not be able to be managed by the getEffectiveModel type helper and so you'd have to check for both the decorator and the `getEffectiveModel`

Reply:
So I updated this based on a suggestion from @johanste. If we have things in an intersection and one is from the service namespace, and the other is not, we prefer the thing from the service namespace. Not sure if we actually want to do that, but it seems more reasonable than just arbitrarily choosing the first thing.

One thing to note is that this doesn't work with spread, which I'm currently looking at.